src/LibVLCSharp.Forms/Shared/PlaybackControls.xaml.cs

```diff
@@ -821,14 +813,113 @@ private void OnAudioTracksChanged(object sender, EventArgs e)
+
+        private void TracksButton_Clicked(object sender, EventArgs e)
+        {
+            if (ControlsPanel != null)
+                ControlsPanel.IsVisible = false;
+            var audioTrackSouces = LoadTracks(Manager.Get<AudioTracksManager>());
+            var videoTrackSouces = LoadTracks(Manager.Get<VideoTracksManager>());
+            var subtitleTracksSources = LoadTracks(Manager.Get<SubtitlesTracksManager>());
+            if (AudioTracksListView != null && VideoTracksListView != null && SubtitlesTracksListView != null)
+            {
+                AudioTracksListView.ItemsSource = audioTrackSouces;
+                VideoTracksListView.ItemsSource = videoTrackSouces;
+                SubtitlesTracksListView.ItemsSource = subtitleTracksSources;
+            }
+
+            if (TracksOverlayView != null)
+                TracksOverlayView.IsVisible = true;
+        }
+
+        private void AudioTracksItemTapped(object sender, ItemTappedEventArgs e)
+        {
+            var track = (TrackViewModel)e.Item;
+            var manager = Manager.Get<AudioTracksManager>();
+            SelectTrack(manager, track, AudioTracksListView);
+        }
+
+        private void VideoTracksItemTapped(object sender, ItemTappedEventArgs e)
+        {
+            var track = (TrackViewModel)e.Item;
+            var manager = Manager.Get<VideoTracksManager>();
+            SelectTrack(manager, track, VideoTracksListView);
+        }
+
+        private void SubtitlesTracksItemTapped(object sender, ItemTappedEventArgs e)
+        {
+            var track = (TrackViewModel)e.Item;
+            var manager = Manager.Get<SubtitlesTracksManager>();
+            SelectTrack(manager, track, SubtitlesTracksListView);
+        }
+
+        private void SelectTrack(TracksManager manager, TrackViewModel track, ListView? tracksListview)
+        {
+            var tracks = manager.Tracks;
+            var currentTrackId = manager.CurrentTrackId;
+            if (tracks == null || track.Selected)
+                return;
+
+            var foundTrack = tracks.FirstOrDefault(t => t.Name == track.Name);
+            if (foundTrack.Name != null)
+            {
+                manager.CurrentTrackId = foundTrack.Id;
+                UpdateTracksListviewItemsSource(track, tracksListview);
+            }
+            else
+            {
+                manager.CurrentTrackId = -1;
+            }
+        }
+
+        private void UpdateTracksListviewItemsSource(TrackViewModel selectedTrack, ListView? trackListView)
+        {
+            if (trackListView != null)
+            {
+                var itemSources = (ObservableCollection<TrackViewModel>)trackListView.ItemsSource;
+                var previousTrack = itemSources.FirstOrDefault(t => t.Selected);
+
+                itemSources.ElementAt(previousTrack.Id).Selected = false;
+                itemSources.ElementAt(selectedTrack.Id).Selected = true;
```

Comment:
```suggestion
                previousTrack.Selected = false;
                selectedTrack.Selected = true;
```

You already have the viewModel object references, you don't need to perform another lookup in the object.

Reply:
If Selected is not updated here, It will always remain false because It is used to hide or display the check buttons on the ListView.

Reply:
I'm not telling you not to set the Selected property, I'm telling you that you don't need to search for things in the itemsSource (espacially based on the index), because you already have the ViewModel instance

Reply:
Ooh sorry. You're right. I just removed it.